_rules/aria-required-accessible-name-gp8n89.md

```diff
@@ -0,0 +1,278 @@
+
+Each target element has an [accessible name][] that is not empty (`""`).
+
+## Assumptions
+
+There are no assumptions.
+
+## Accessibility Support
+
+Elements with [explicit semantic role][] of `heading` and with no [accessible name][] seem to be consistently ignored by assistive technologies. Nonetheless, in the context of this rule, they fail since [WAI-ARIA 1.2][wai-aria 1.2] mandates the presence of an [accessible name][].
+
+The [accessible name and description computation][] suggests that if an `aria-labelledby` attribute refers to an existing but empty element, the computation should stop and return an empty name without defaulting to the next steps. Several user agents and assistive technologies chose to use the next step in the computation in this case (ultimately defaulting to the content).
+
+## Background
+
+The applicability of this rule is limited to elements that have an [explicit semantic role][] that is not identical to their [implicit semantic role][]. Both [WAI-ARIA 1.2][wai-aria 1.2] and [ARIA in HTML](https://www.w3.org/TR/html-aria/#docconformance) discourage authors from using ARIA when the host language provides a feature with equivalent role semantics and values.
+
+An example of an element that has an [implicit semantic role][] that is identical to its [explicit semantic role][] is a `<table role="table">` element. These elements are not applicable because they have extra requirements and should thus be checked separately.
+
+Elements subjected to the [Presentational Roles Conflict Resolution][] are not covered by this rule, since their [explicit semantic role][] of `none` or `presentation` do not require an accessible name. However, if the [Presentational Roles Conflict Resolution][] is invoked, although the rule doesn't apply, an accessible name might still be necessary.
+
+### Bibliography
+
+- [WAI-ARIA 1.2][wai-aria 1.2]
+- [ARIA in HTML](https://www.w3.org/TR/html-aria/#docconformance)
+- [Understanding Success Criterion 1.3.1: Info and Relationships](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value)
+- [Understanding Success Criterion 4.1.2: Name, Role, Value](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The `div` element with role `button` has an [accessible name][] provided by its content.
+
+```html
+<div role="button" tabindex="0">Submit</div>
+```
+
+#### Passed Example 2
+
+The `div` element with role `img` has an [accessible name][] provided by the `aria-label` [attribute value][].
+
+```html
+<div role="img" aria-label="Rating: 5 out of 5 stars">
+  <span aria-hidden="true">★★★★★</span>
+</div>
+```
+
+#### Passed Example 3
+
+The `div` element with role `checkbox` has an [accessible name][] provided thanks to the `aria-labelledby` attribute.
+
+```html
+<div role="checkbox" aria-checked="false" tabindex="0" aria-labelledby="passed-example-3"></div>
+<div id="passed-example-3">I agree with terms and conditions</div>
```

Comment:
I'd recommend against using `passed-example-3` as an `id` to limit the changes needed if we add more examples.
(also applies to some other examples)